dev_guide/builds.adoc

```diff
@@ -1426,6 +1432,94 @@ Configuration change triggers currently only work when creating a new
 ====
 
+[[build-hooks]]
+
+== Build Hooks
+
+Build hooks can be used as a part of the build flow to run tests inside a
+temporary container and validate your image before it is being pushed to a
+registry. The command is executed with the current working directory set to the
+image's `*WORKDIR*` which is your current working directory where your code is
+located. The hook will fail if the script or command returns a non-zero exit
+code or if there's any other error related to starting the temporary container.
+When the hook fails it will mark the build as failed.
+====
+[source,yaml]
+spec:
+  postCommit:
+    script: "rake test" <1>
+
+<1> `script` is a shell script to be run with `/bin/sh -c`. Use this when a
+shell script is appropriate to execute the post build hook, for example for
+running unit tests with "rake test". If you need control over the image
+entrypoint, or if the image does not have "/bin/sh", use `command` and/or
+`args`.
+
+[source,yaml]
+spec:
+  postCommit:
+    command: ["/bin/sh", "-c"] <2>
+    args: ["rake, ,"test"] <3>
+
+<2> `command` is the command to run. This might be needed if the image doesn't
+have "/bin/sh" or if you do not want to use a shell. In all other cases, using
+`script` might be more convenient.
+<3> `args` is a list of arguments that are provided to either `command`,
+`script` or the default entrypoint of the image. The arguments are placed
+immediately after the command to be run.
+
+
+  NOTE: It is invalid to provide both `Script` and `Command` simultaneously. If
+  none of the fields are specified, the hook is not executed.
+====
+
+There are five different ways to configure the post build hook. As an example,
+all forms below are equivalent and will execute `rake test --verbose`.
+
+1. Shell script:
+[source,yaml]
+postCommit:
+  script: "rake test --verbose"
+
+2. Command as the image entrypoint:
+[source,yaml]
+postCommit:
+  command: ["rake", "test", "verbose"]
+
+* `command` overrides the image entrypoint in the exec form, as documented in
+Docker: https://docs.docker.com/engine/reference/builder/#entrypoint.
+
+3. Pass arguments to the default entrypoint:
+[source,yaml]
+postCommit:
+  args: ["rake", "test", "--verbose"]
+
+* This form is only useful if the image entrypoint can handle arguments.
+
+4. Shell script with arguments:
+[source,yaml]
+postCommit:
+  script: "rake test"
+  args: ["--verbose"]
+
+ * This form is useful if you need to pass arguments that would otherwise be
+ hard to quote properly in the shell script. In the script, $0 will be
+ "/bin/sh" and $1, $2, etc, are the positional arguments from `args`.
+
+ 5. Command with arguments:
+[source,yaml]
+postCommit:
+  command: ["rake", "test"]
+  args: ["--verbose"]
+
+ * This form is equivalent to appending the arguments to `command`.
+
+The post commit hook is not limited to running tests but can be used for other
+commands as well, what you have to bare in mind is that changes made by the
```

Comment:
"comands as well.  It is important to be aware that changes made by the post-build command will not be persisted since it runs in a temporary container.  As such, no actions performed by the post-build command will affect the final image.  This behavior allows for, among other uses, the installation and usage of test dependencies without affecting the final image."